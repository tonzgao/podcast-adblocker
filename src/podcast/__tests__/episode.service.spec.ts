import { Test, TestingModule } from '@nestjs/testing';
import { ConfigModule } from '@nestjs/config';

import { EpisodeService } from '../episode.service';

describe('EpisodeService', () => {
  let episodeService: EpisodeService;

  beforeEach(async () => {
    const app: TestingModule = await Test.createTestingModule({
      imports: [ConfigModule.forRoot({ isGlobal: true })],
      controllers: [],
      providers: [EpisodeService],
    }).compile();

    episodeService = app.get<EpisodeService>(EpisodeService);
  });

  xit('should download and serve audio files', async () => {
    expect(
      await episodeService.handle(
        'https://file-examples-com.github.io/uploads/2017/11/file_example_MP3_700KB.mp3',
        // eslint-disable-next-line @typescript-eslint/no-empty-function
        { set: () => {} },
      ),
    ).toMatchSnapshot();
  });
});
