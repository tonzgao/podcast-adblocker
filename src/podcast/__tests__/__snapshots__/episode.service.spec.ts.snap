// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeService root should download and serve audio files 1`] = `
StreamableFile {
  "options": Object {},
  "stream": ReadStream {
    "_events": Object {
      Symbol(kConstruct): [Function],
    },
    "_eventsCount": 1,
    "_maxListeners": undefined,
    "_readableState": ReadableState {
      "autoDestroy": true,
      "awaitDrainWriters": null,
      "buffer": BufferList {
        "head": null,
        "length": 0,
        "tail": null,
      },
      "closeEmitted": false,
      "closed": false,
      "constructed": false,
      "dataEmitted": false,
      "decoder": null,
      "defaultEncoding": "utf8",
      "destroyed": false,
      "emitClose": true,
      "emittedReadable": false,
      "encoding": null,
      "endEmitted": false,
      "ended": false,
      "errorEmitted": false,
      "errored": null,
      "flowing": null,
      "highWaterMark": 65536,
      "length": 0,
      "multiAwaitDrain": false,
      "needReadable": false,
      "objectMode": false,
      "pipes": Array [],
      "readableListening": false,
      "reading": false,
      "readingMore": false,
      "resumeScheduled": false,
      "sync": true,
      Symbol(kPaused): null,
    },
    "bytesRead": 0,
    "closed": false,
    "end": Infinity,
    "fd": null,
    "flags": "r",
    "mode": 438,
    "path": "/tmp/file_example_MP3_700KB.mp3",
    "pos": undefined,
    "start": undefined,
    Symbol(kFs): Object {
      "Dir": [Function],
      "Dirent": [Function],
      "F_OK": 0,
      "FileReadStream": [Function],
      "FileWriteStream": [Function],
      "R_OK": 4,
      "ReadStream": [Function],
      "Stats": [Function],
      "W_OK": 2,
      "WriteStream": [Function],
      "X_OK": 1,
      "_toUnixTimestamp": [Function],
      "access": [Function],
      "accessSync": [Function],
      "appendFile": [Function],
      "appendFileSync": [Function],
      "chmod": [Function],
      "chmodSync": [Function],
      "chown": [Function],
      "chownSync": [Function],
      "close": [Function],
      "closeSync": [Function],
      "constants": Object {
        "COPYFILE_EXCL": 1,
        "COPYFILE_FICLONE": 2,
        "COPYFILE_FICLONE_FORCE": 4,
        "F_OK": 0,
        "O_APPEND": 1024,
        "O_CREAT": 64,
        "O_DIRECT": 16384,
        "O_DIRECTORY": 65536,
        "O_DSYNC": 4096,
        "O_EXCL": 128,
        "O_NOATIME": 262144,
        "O_NOCTTY": 256,
        "O_NOFOLLOW": 131072,
        "O_NONBLOCK": 2048,
        "O_RDONLY": 0,
        "O_RDWR": 2,
        "O_SYNC": 1052672,
        "O_TRUNC": 512,
        "O_WRONLY": 1,
        "R_OK": 4,
        "S_IFBLK": 24576,
        "S_IFCHR": 8192,
        "S_IFDIR": 16384,
        "S_IFIFO": 4096,
        "S_IFLNK": 40960,
        "S_IFMT": 61440,
        "S_IFREG": 32768,
        "S_IFSOCK": 49152,
        "S_IRGRP": 32,
        "S_IROTH": 4,
        "S_IRUSR": 256,
        "S_IRWXG": 56,
        "S_IRWXO": 7,
        "S_IRWXU": 448,
        "S_IWGRP": 16,
        "S_IWOTH": 2,
        "S_IWUSR": 128,
        "S_IXGRP": 8,
        "S_IXOTH": 1,
        "S_IXUSR": 64,
        "UV_DIRENT_BLOCK": 7,
        "UV_DIRENT_CHAR": 6,
        "UV_DIRENT_DIR": 2,
        "UV_DIRENT_FIFO": 4,
        "UV_DIRENT_FILE": 1,
        "UV_DIRENT_LINK": 3,
        "UV_DIRENT_SOCKET": 5,
        "UV_DIRENT_UNKNOWN": 0,
        "UV_FS_COPYFILE_EXCL": 1,
        "UV_FS_COPYFILE_FICLONE": 2,
        "UV_FS_COPYFILE_FICLONE_FORCE": 4,
        "UV_FS_O_FILEMAP": 0,
        "UV_FS_SYMLINK_DIR": 1,
        "UV_FS_SYMLINK_JUNCTION": 2,
        "W_OK": 2,
        "X_OK": 1,
      },
      "copyFile": [Function],
      "copyFileSync": [Function],
      "cp": [Function],
      "cpSync": [Function],
      "createReadStream": [Function],
      "createWriteStream": [Function],
      "exists": [Function],
      "existsSync": [Function],
      "fchmod": [Function],
      "fchmodSync": [Function],
      "fchown": [Function],
      "fchownSync": [Function],
      "fdatasync": [Function],
      "fdatasyncSync": [Function],
      "fstat": [Function],
      "fstatSync": [Function],
      "fsync": [Function],
      "fsyncSync": [Function],
      "ftruncate": [Function],
      "ftruncateSync": [Function],
      "futimes": [Function],
      "futimesSync": [Function],
      "lchmod": undefined,
      "lchmodSync": undefined,
      "lchown": [Function],
      "lchownSync": [Function],
      "link": [Function],
      "linkSync": [Function],
      "lstat": [Function],
      "lstatSync": [Function],
      "lutimes": [Function],
      "lutimesSync": [Function],
      "mkdir": [Function],
      "mkdirSync": [Function],
      "mkdtemp": [Function],
      "mkdtempSync": [Function],
      "open": [Function],
      "openSync": [Function],
      "opendir": [Function],
      "opendirSync": [Function],
      "promises": Object {
        "access": [Function],
        "appendFile": [Function],
        "chmod": [Function],
        "chown": [Function],
        "copyFile": [Function],
        "cp": [Function],
        "lchmod": [Function],
        "lchown": [Function],
        "link": [Function],
        "lstat": [Function],
        "lutimes": [Function],
        "mkdir": [Function],
        "mkdtemp": [Function],
        "open": [Function],
        "opendir": [Function],
        "readFile": [Function],
        "readdir": [Function],
        "readlink": [Function],
        "realpath": [Function],
        "rename": [Function],
        "rm": [Function],
        "rmdir": [Function],
        "stat": [Function],
        "symlink": [Function],
        "truncate": [Function],
        "unlink": [Function],
        "utimes": [Function],
        "watch": [Function],
        "writeFile": [Function],
      },
      "read": [Function],
      "readFile": [Function],
      "readFileSync": [Function],
      "readSync": [Function],
      "readdir": [Function],
      "readdirSync": [Function],
      "readlink": [Function],
      "readlinkSync": [Function],
      "readv": [Function],
      "readvSync": [Function],
      "realpath": [Function],
      "realpathSync": [Function],
      "rename": [Function],
      "renameSync": [Function],
      "rm": [Function],
      "rmSync": [Function],
      "rmdir": [Function],
      "rmdirSync": [Function],
      "stat": [Function],
      "statSync": [Function],
      "symlink": [Function],
      "symlinkSync": [Function],
      "truncate": [Function],
      "truncateSync": [Function],
      "unlink": [Function],
      "unlinkSync": [Function],
      "unwatchFile": [Function],
      "utimes": [Function],
      "utimesSync": [Function],
      "watch": [Function],
      "watchFile": [Function],
      "write": [Function],
      "writeFile": [Function],
      "writeFileSync": [Function],
      "writeSync": [Function],
      "writev": [Function],
      "writevSync": [Function],
    },
    Symbol(kIsPerformingIO): false,
    Symbol(kCapture): false,
  },
}
`;
